models/Asset/Service.php

```diff
@@ -78,6 +78,12 @@ public function copyRecursive($target, $source)
 
         $source->getProperties();
 
+        // triggers actions before asset cloning
+        $event = new AssetEvent($source, [
+            'target_element' => $target,
+        ]);
+        \Pimcore::getEventDispatcher()->dispatch($event, AssetEvents::PRE_COPY);
```

Comment:
we have to get the `$target` from event argument `$target = $event->getArgument('target_element');` after event dispatch, so that the latest one is used afterwards.